src/Product/ProductSyncer.php

```diff
@@ -357,10 +356,25 @@ protected function handle_delete_errors( array $invalid_products ) {
 	 * @throws ProductSyncerException If the Google Merchant Center connection is not ready.
 	 */
 	protected function validate_merchant_center_setup(): void {
-		if ( ! $this->merchant_center->is_ready_for_syncing() ) {
+		if ( ! $this->merchant_center->is_ready() ) {
 			do_action( 'woocommerce_gla_error', 'Cannot sync any products before setting up Google Merchant Center.', __METHOD__ );
 
 			throw new ProductSyncerException( __( 'Google Merchant Center has not been set up correctly. Please review your configuration.', 'google-listings-and-ads' ) );
 		}
+
+		if ( ! $this->merchant_center->should_sync() ) {
+			do_action(
+				'woocommerce_gla_error',
+				'Cannot sync any products because they are being fetched automatically.',
+				__METHOD__
+			);
+
+			throw new ProductSyncerException(
+				__(
+					'Manual product sync will not run if the automatic data fetching is enabled. Please review your configuration in Google Listing and Ads settings.',
+					'google-listings-and-ads'
+				)
+			);
+		}
```

Comment:
I wanted to test these errors, but I noticed that `validate_merchant_center_setup` is only triggered when the product syncers job is processing the items. So theoretically, we shouldn't encounter this error. However, I think it's a good idea to keep it just in case. While reviewing the code, I saw that for the notification part, we don't have anything similar to "validate_merchant_center_setup". We only check if the notification service is enabled, but not the MC setup. should we add something similar to the notification? Or am I missing something?

https://github.com/woocommerce/google-listings-and-ads/blob/1f3a18057ee27c08ae10261a1ab9b9adbf17a6df/src/Jobs/Notifications/AbstractNotificationJob.php#L88

Reply:
> I wanted to test these errors, but I noticed that validate_merchant_center_setup is only triggered when the product syncers job is processing the items. So theoretically, we shouldn't encounter this error.

You can trigger it in ConnectionTest Page for example

Reply:
>  We only check if the notification service is enabled, but not the MC setup. should we add something similar to the notification? Or am I missing something?

We can do so in a follow up. I added the item to the list.